packages/teams-js/src/public/meeting.ts

```diff
@@ -925,6 +925,124 @@ export namespace meeting {
 
+  /**
+   * This function is used to join a meeting.
+   * This opens a meeting in a new window for the desktop app.
+   * In case of a web app, it will close the current app and open the meeting in the same tab.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @throws error if the meeting join fails, the promise will reject to an object with the error message.
+   */
+
+  export function joinMeeting(joinMeetingParams: JoinMeetingParams): Promise<void> {
+    if (!validateJoinMeetingParams(joinMeetingParams)) {
+      return Promise.reject('Invalid joinMeetingParams');
+    }
+    const { threadId, messageId, replyChainMessageId } = joinMeetingParams.chatInfo;
+
+    joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl);
+    const serializedJoinMeetingParams = {
+      joinWebUrl: joinMeetingParams.joinWebUrl.href,
+      subject: joinMeetingParams.subject,
+      source: joinMeetingParams.source || EventActionSource.Other,
+      chatInfo: {
+        threadId,
+        messageId: messageId || '0',
+        replyChainMessageId: replyChainMessageId || '0',
+      },
+    };
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime);
+      resolve(
+        sendAndHandleStatusAndReason(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Meeting_JoinMeeting),
+          'meeting.joinMeeting',
+          serializedJoinMeetingParams,
+        ),
+      );
+    });
+  }
+
+  /**
+   * This function is used to check the validity of joinMeetingParams which return true if the joinMeetingParams' joinWebUrl and chatInfo are valid else false.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @returns false if joinMeetingParams' joinWebUrl or chatInfo is not valid else true.
+   */
+  function validateJoinMeetingParams(joinMeetingParams: JoinMeetingParams): boolean {
+    if (
+      !joinMeetingParams?.chatInfo?.threadId ||
+      !joinMeetingParams?.joinWebUrl ||
+      joinMeetingParams.chatInfo.threadId === ''
+    ) {
+      return false;
+    }
+    if (typeof joinMeetingParams.joinWebUrl === 'string' && joinMeetingParams.joinWebUrl === '') {
+      return false;
+    }
+
+    if (joinMeetingParams.joinWebUrl instanceof URL && joinMeetingParams.joinWebUrl.href === '') {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Contains information associated with parameters required for joining the Microsoft Teams meetings.
+   * More details can be found at:
+   * [Online Meeting Base - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/onlinemeetingbase?view=graph-rest-1.0)
+   */
+  export interface JoinMeetingParams {
+    /** The join URL of the online meeting. */
+    joinWebUrl: URL | string;
+    /** The chat information associated with this online meeting. {@link ChatInfo}*/
+    chatInfo: ChatInfo;
+    /** The subject of the meeting. */
+    subject?: string;
```

Comment:
Based on the other comment you wrote, I recommend updating the property name and description. Calling this `subject` is likely to make a caller think they need to find the right subject for the meeting before populating this (or worse, given that it's now optional, just never populate it at all because it's too much work to find the subject, at which point why even allow this to be passed).

Consider calling it something like `temporaryMeetingWindowTitle`

Reply:
@AE-MS, we have been following the public documentation: https://learn.microsoft.com/en-us/graph/api/resources/onlinemeetingbase?view=graph-rest-1.0, on @TrevorJoelHarris suggestions and found that subject is the field name which we would have to go with. Hence the name and description.